src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -712,7 +712,7 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
 
             final EnvVars environment = project instanceof AbstractProject ? GitUtils.getPollEnvironment((AbstractProject) project, workspace, launcher, listener, false) : new EnvVars();
 
-            GitClient git = createClient(listener, environment, project, Jenkins.get(), null);
+            GitClient git = createClient(listener, environment, project, project.getLastBuild(), Jenkins.get(), null);
```

Comment:
I assume this is calling `project.getLastBuild()` rather than using the existing `lastBuild` variable because there may be a more recent `lastBuild` that has become available.  Is that assumption correct or is there another reason to not use the existing `lastBuild` that is referenced earlier in the method?

Reply:
No reason other than I just did not notice that such a variable existed. For purposes of https://github.com/jenkinsci/github-branch-source-plugin/pull/527 it does not matter—any build of the `Job` (indeed, for multibranch, any build of any sibling branch project!) would do.

Thanks, it think this will allow me to clean up somewhat.